### PR TITLE
onetbb: fixed hwloc setting on Windows

### LINKED ISSUE #19168: [package] onetbb/2021.3.0: Invalid character escape '\l'
### Description

For details on this, see : https://github.com/conan-io/conan/issues/10539

`conan_toolchain.cmake` is broken due to unescaped backslashes on windows. See log.

### Package and Environment Details

* Package Name/Version: **onetbb/2021.3.0#e447f12564bd0a5df3f2bd5781cf5cc7**
* Operating System+version: **Windows 11**
* Compiler+version: **MSVC 2022**
* Docker image: **?**
* Conan version: **conan 2.0.9**
* Python version: **Python 3.7.9**

### Conan profile

======== Input profiles ========
Profile host:
[settings]
arch=x86
build_type=Debug
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=193
os=Windows

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows

### Steps to reproduce

Install (build) onetbb/2021.3.0#e447f12564bd0a5df3f2bd5781cf5cc7 with cmakedeps / cmaketoolchain generator.

### Logs

<details><summary>Click to expand log</summary>

```
CMake Error at C:/Users/pgroarke/.conan2/p/b/onetb3e83bbcb22a9a/b/build/generators/conan_toolchain.cmake:117 (set):
  Syntax error in cmake code at

    C:/Users/pgroarke/.conan2/p/b/onetb3e83bbcb22a9a/b/build/generators/conan_toolchain.cmake:117

  when parsing string

    C:/Users/pgroarke/.conan2/p/b/hwloc93107a9ec9dfe/p\lib\hwloc.lib

  Invalid character escape '\l'.
```

</details>

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Specify library name and version:  **onetbb/2021.10.0**

Closes #19168

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/onetbb/all/conanfile.py [hunk 1]
```diff
@@ -129,13 +129,16 @@ def generate(self):
...
+            toolchain.variables[f"CMAKE_HWLOC_{self._tbbbind_hwloc_version}_LIBRARY_PATH"] = \
+                os.path.join(hwloc_package_folder, "lib", hwloc_lib_name).replace("\\", "/")
+            toolchain.variables[f"CMAKE_HWLOC_{self._tbbbind_hwloc_version}_INCLUDE_PATH"] = \
+                os.path.join(hwloc_package_folder, "include").replace("\\", "/")
             if self.settings.os == "Windows":
-                toolchain.variables[f"CMAKE_HWLOC_{self._tbbbind_hwloc_version}_DLL_PATH"] = os.path.join(hwloc_package_folder, "bin", "hwloc.dll")
-        toolchain.generate()
+                toolchain.variables[f"CMAKE_HWLOC_{self._tbbbind_hwloc_version}_DLL_PATH"] = \
+                    os.path.join(hwloc_package_folder, "bin", "hwloc.dll").replace("\\", "/")
+            toolchain.generate()
```
Comment (SpaceIm) [inline recipes/onetbb/all/conanfile.py:141]:
```suggestion
        toolchain.generate()
```

**Comment by SpaceIm** [comment]
For readability could you move generation of pkgconfigdeps after cmaketoolchain logic please. It's confusing with this unecessary interleaved logic.

File: recipes/onetbb/all/conanfile.py (hunk 1 above)
Comment (ilya-lavrenov) [reply recipes/onetbb/all/conanfile.py]: fixed

**Comment by ilya-lavrenov** [comment]
@RubenRBS @SSE4 could you please review?

**Review by SSE4 — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@RubenRBS could you please review?

**Comment by ilya-lavrenov** [comment]
@RubenRBS please, have a look

**Review by RubenRBS — APPROVED** [review]
I see where the error comes from, nice catch!

**Pull request merged**